### DISCUSSION TITLE
how to make an custom paragraph block

### DISCUSSION THREAD

**Discussion by rien7** [toeverything/blocksuite#6549] [Q&A] [answered]
If I want to make an custom paragraph block similar to `affine:paragraph`, where should I start from?

`affine:note` not accept custom block as its child, should I build to editor from scratch?

**Comment by doodlewind** [comment, marked as answer]
If you are going to build an *embed block*, the API surface is more polished currently ([docs](https://blocksuite.io/guide/working-with-block-tree.html#defining-new-blocks)), and `affine:note` supports `embed-*` blocks as is children.

But regarding paragraph block, it's kind of "built-in" for default editors for AFFiNE, so there would be more edge cases (e.g., in selection) that should be handled if you are going to overwrite it with your own text block type. It might be more efficient if you can propose your feature request for your customization needs, and we can see if there could be more exposed API for the current paragraph block to fit.
